src/osd/SnapMapper.cc

```diff
@@ -583,15 +575,49 @@ int SnapMapper::get_next_objects_to_trim(
       out->push_back(next_decoded.second);
       pos = next.first;
     }
+
+    if (out->size() >= max) {
+      return;
+    }
```

Comment:
You can reach that point in two conditions, size and changing-itr.
The first one should return, the second should inc the iterator

Reply:
Oh, so this can be also added to the outer loop condition, right?

Reply:
Do you want to check if (out->size() >= max)  is the outer loop?

Reply:
> Do you want to check if (out->size() >= max) is the outer loop?

I understand now that this will increase the prefix_itr, this return is ok as is.

Reply:
no, it will increase the iterator to next one before the current itr is depleted